include/vcpkg/base/files.h

```diff
@@ -33,41 +19,79 @@ namespace vcpkg
         std::error_code ec;
     };
 
-#if defined(_WIN32)
-    struct IsSlash
+    enum class CopyOptions
     {
-        bool operator()(const wchar_t c) const noexcept { return c == L'/' || c == L'\\'; }
+        none = 0,
+
+        existing_mask = 0xF,
+        skip_existing = 0x1,
+        overwrite_existing = 0x2,
+        update_existing = 0x4,
+
+        recursive = 0x10,
+
+        symlinks_mask = 0xF00,
+        copy_symlinks = 0x100,
+        skip_symlinks = 0x200,
     };
-#else
-    struct IsSlash
+
+    struct Path
     {
-        bool operator()(const char c) const noexcept { return c == '/'; }
-    };
-#endif
+        Path() : m_str() { }
+        Path(const Path&) = default;
+        Path(Path&&) = default;
+        Path& operator=(const Path&) = default;
+        Path& operator=(Path&&) = default;
+
+        Path(const StringView sv) : m_str(sv.to_string()) { }
+        Path(const std::string& s) : m_str(s) { }
+        Path(std::string&& s) : m_str(std::move(s)) { }
+        Path(const char* s) : m_str(s) { }
+        template<class Iter>
+        Path(Iter first, Iter last) : m_str(first, last)
+        {
+        }
+        Path(const char* first, size_t size) : m_str(first, size) { }
 
-    constexpr IsSlash is_slash;
+        const std::string& native() const noexcept { return m_str; }
+        std::string& native() noexcept { return m_str; }
+        operator StringView() const noexcept { return m_str; }
 
-    using stdfs::copy_options;
-    using stdfs::file_status;
-    using stdfs::file_type;
-    using stdfs::path;
+        const char* c_str() const noexcept { return m_str.c_str(); }
 
-    path u8path(StringView s);
-    inline path u8path(const char* first, const char* last) { return u8path(StringView{first, last}); }
-    inline path u8path(std::initializer_list<char> il) { return u8path(StringView{il.begin(), il.end()}); }
-    inline path u8path(const char* s) { return u8path(StringView{s, s + ::strlen(s)}); }
+        std::string generic_u8string() const;
 
-    inline path u8path(std::string::const_iterator first, std::string::const_iterator last)
-    {
-        auto firstp = &*first;
-        return u8path(StringView{firstp, firstp + (last - first)});
-    }
+        bool empty() const noexcept { return m_str.empty(); }
 
-    std::string u8string(const path& p);
-    std::string generic_u8string(const path& p);
+        Path operator/(StringView sv) const;
+        Path operator+(StringView sv) const;
 
-    // equivalent to p.lexically_normal()
-    path lexically_normal(const path& p);
+        Path& operator/=(StringView sv);
+        Path& operator+=(StringView sv);
+
+        void replace_filename(const Path&);
+        void remove_filename();
+        Path as_preferred();
+        void clear() { m_str.clear(); }
+        Path lexically_normal() const;
```

Comment:
```suggestion
        Path lexically_normal() const;
        void lexically_normalize();
```

Reply:
The lexically_normal algorithm cannot be done in place so I'm not sure there's much to gain here

Reply:
Oh, it can't? Isn't the result strictly shorter than the input?